### PR TITLE
Fix goto-def not working on the last character in a document

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1615061/

### CODE REVIEW

File: src/EditorFeatures/Core/GoToDefinition/GoToDefinitionCommandHandler.cs [hunk 1]
```diff
@@ -74,14 +74,20 @@ public bool ExecuteCommand(GoToDefinitionCommandArgs args, CommandExecutionConte
...
 
+            // If the file is empty, there's nothing to be on that we can goto-def on.  This also ensures that we can
+            // create an appropriate non-empty tracking span later on.
+            var currentSnapshot = subjectBuffer.CurrentSnapshot;
+            if (currentSnapshot.Length == 0)
+                return false;
+
             Contract.ThrowIfNull(document);
-            var caretPos = args.TextView.GetCaretPoint(subjectBuffer);
+            var caretPos= args.TextView.GetCaretPoint(subjectBuffer);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/GoToDefinition/GoToDefinitionCommandHandler.cs:90]:
```suggestion
            var caretPos = args.TextView.GetCaretPoint(subjectBuffer);
```

**Comment by CyrusNajmabadi** [comment]
@ryzngard ptal.

**Review by genlu — APPROVED** [review]
Change LGTM. Could we add a test to cover this, maybe with top level statement?

**Comment by CyrusNajmabadi** [comment]
>Change LGTM. Could we add a test to cover this, maybe with top level statement?

sure.  i can do that :)

**Review by ryzngard — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Test coming in next PR.

**Pull request merged**